subprojects/analysis-kotlin-descriptors/compiler/src/main/kotlin/org/jetbrains/dokka/analysis/kotlin/descriptors/compiler/impl/KotlinSampleProvider.kt

```diff
@@ -0,0 +1,154 @@
+package org.jetbrains.dokka.analysis.kotlin.descriptors.compiler.impl
+
+import com.intellij.psi.PsiElement
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.runBlocking
+import org.jetbrains.dokka.DokkaConfiguration
+import org.jetbrains.dokka.InternalDokkaApi
+import org.jetbrains.dokka.analysis.kotlin.descriptors.compiler.CompilerDescriptorAnalysisPlugin
+import org.jetbrains.dokka.analysis.kotlin.descriptors.compiler.KDocFinder
+import org.jetbrains.dokka.analysis.kotlin.descriptors.compiler.configuration.SamplesKotlinAnalysis
+import org.jetbrains.dokka.plugability.DokkaContext
+import org.jetbrains.dokka.plugability.plugin
+import org.jetbrains.dokka.plugability.querySingle
+import org.jetbrains.kotlin.analysis.kotlin.internal.SampleProvider
+import org.jetbrains.kotlin.analysis.kotlin.internal.SampleProviderFactory
+import org.jetbrains.kotlin.name.FqName
+import org.jetbrains.kotlin.psi.KtBlockExpression
+import org.jetbrains.kotlin.psi.KtDeclarationWithBody
+import org.jetbrains.kotlin.psi.KtFile
+import org.jetbrains.kotlin.resolve.lazy.ResolveSession
+
+class KotlinSampleProviderFactory(val context: DokkaContext): SampleProviderFactory {
+    override fun build(): SampleProvider {
+        return KotlinSampleProvider(context)
+    }
+
+}
+/**
+ * It's declared as open since StdLib has its own sample transformer
+ * with [processBody] and [processImports]
+ */
```

Comment:
Oh, I didn't know stdlib actually used something like this... 

It can be used as a very short-term solution that unblocks stdlib, but it's definitely not OK for stdlib to depend on internal API, let alone on one of the analysis modules.

Would you be able to research what exactly the use case in stdlib is, and to think of a stable alternative to go into `kotlin-analysis-api`? As a separate issue / PR, of course

Reply:
https://github.com/JetBrains/kotlin/blob/master/libraries/tools/kotlin-stdlib-docs/plugins/dokka-samples-transformer-plugin/src/main/kotlin/org/jetbrains/dokka/kotlinlang/KotlinWebsiteSamplesTransformer.kt
It transforms `assert*` to `println`.
The easiest way to move the transformer to Dokka, but it needs discussion.